### DISCUSSION TITLE
Spotify Connect volume too loud

### DISCUSSION THREAD

**Discussion by JK3P** [rern/rAudio#1806] [General]
New to rAudio, impressed and pleased so far. Works like a breeze listening webradio and local files.

Using Spotify Connect blows volume level to hearing damage...
Has anybody an idea to let spotify connect behave volume-wise equal to listening webradio and local files?

Setup:
Using HifiBerry AMP2 on Rpi 3B+ on rAudio 20241025. Mixer Device = Digital and Volume Control = DAC Hardware (mixer device)

Thanks in advance, greetings from Netherlands,
Joost

**Comment by AndyFx12** [comment]
Hi Joost,

at rAudio, spotify's volume is controlled by your e. g. Spotify-App. If you change your volume in you Spotify-App, rAudio's volume will follow.

regards, Andy

**Reply by JK3P** [reply]
Hi Andy,

Thanx, I'm aware of controlling spotify's volume by using slider within the Spotify-App. Let me explain what's happening usung Spotify connect on rAudio.

When connecting to Spotiftd@rAudio the volume on RAudio boost to 68 where the workable volume range on amp2 and my speakers is roughly between 0 and 40.
The volume slider condition is about 25% wihin the app, reducing this to 'zero' resulting the rAudio volume minimises to 37, unfortunately way too loud.

I found out that changing the line 'volume-controller' from 'alsa' to 'alsa_linear' in /etc/spotifyd.conf seems to slove my problem.
Unfortunately this change isn't persistant: after a reboot the volume-controller is reset to 'alsa'.

regards, Joost

**Comment by AndyFx12** [comment]
At latest update, spotify-mixer can be set to "alsa_linear".

regards, Andy

**Comment by JK3P** [comment]
Thanx for your reply. Just tested and works like a charm on the AMP2.
FYI; My faulty SD-card was to blame: the update from 20240818 to 20241025 was also not persistant.
Despite no errors/bad sectors found on the SD-card.

Much compliments to you and your wonderful rAudio:  a revival of my dust collecting AMP2.

Regards, Joost

**Comment by v0d0r** [comment]
I had a similar issue using die DigiAmp+ DAC.  When I play music on the rAudio app from my Nas volume is set to 25% and stays there.  When I connect to rAudio via spotifyd the sound jumps super high.  I will give the above a try but what I have done in the meantime is added `initial_volume = "5"` to /etc/spotifyd.conf   It seems what spotifyd considers 5 is what rAudio is on 25 (if that makes sense?).  Anyway ill see if my hack survives a reboot and also try the linear setting when I have some more time.   BTW.  Thanks rAudio is super cool. I recently migrated from Volumio and was pleasantly surprised how easy r-Audio was.  A few small issues in the transition here and there but found my way quickly. ;-)

**Reply by kan-ibal** [reply]
rAudio is the best :).

**Reply by TinkeringB** [reply]
Thanks!

Had the same problem. 

But, the Linear option under the spotify popup box does not seem to work for me; it jumps back to Default.
<img width="435" height="283" alt="image" src="https://github.com/user-attachments/assets/7c3d02bf-1ac6-4cbf-b885-a371432c390c" />
And, changing "alsa" into "alsa_linear" in /etc/spotifyd.conf resulted in even weirder behaviour. [ticket](https://github.com/Spotifyd/spotifyd/issues/1222)

Thanks @v0d0r 
Adding initial_volume = "5" to /etc/spotifyd.conf  worked wonders :-)

in contrary to what is stated somewhere here above that spotifyd simply follows the setting of the iphone slider initially, after adding this parameter, spotify (?) reduces the volume control slider for this raudio spotify connect on my iphone also; apparently there is 2 way communication about the initial volume setting.
 
Edit: sadly, a pause and play  from the iphone app, sets volume to 93% again :-(

**Reply by TinkeringB** [reply]
and yes -rAudio is the best
